source/Halibut.TestUtils.CompatBinary.Base/DelegateComplexObjectService.cs

```diff
@@ -14,32 +13,37 @@ public DelegateComplexObjectService(IComplexObjectService service)
         {
             this.service = service;
         }
-
-        public ComplexResponse Process(ComplexRequest request)
+        
+        public ComplexObjectMultipleDataStreams Process(ComplexObjectMultipleDataStreams request)
```

Comment:
The duplication of code in this class seems unnecessary:
```
diff --git a/source/Halibut.TestUtils.CompatBinary.Base/DelegateComplexObjectService.cs b/source/Halibut.TestUtils.CompatBinary.Base/DelegateComplexObjectService.cs
index 215f45d..9235ae0 100644
--- a/source/Halibut.TestUtils.CompatBinary.Base/DelegateComplexObjectService.cs
+++ b/source/Halibut.TestUtils.CompatBinary.Base/DelegateComplexObjectService.cs
@@ -16,35 +16,32 @@ public DelegateComplexObjectService(IComplexObjectService service)
         
         public ComplexObjectMultipleDataStreams Process(ComplexObjectMultipleDataStreams request)
         {
-            request.Payload1 = request.Payload1.ConfigureWriterOnReceivedDataStream();
-            request.Payload2 = request.Payload2.ConfigureWriterOnReceivedDataStream();
-
-            var response = service.Process(request);
-            
-            response.Payload1 = response.Payload1.ConfigureWriterOnReceivedDataStream();
-            response.Payload2 = response.Payload2.ConfigureWriterOnReceivedDataStream();
+            var response = service.Process(ConfigureWriterOnReceivedDataStream(request));
+            return ConfigureWriterOnReceivedDataStream(response);
+        }
 
-            return response;
+        static ComplexObjectMultipleDataStreams ConfigureWriterOnReceivedDataStream(ComplexObjectMultipleDataStreams complexObjectMultipleDataStreams)
+        {
+            complexObjectMultipleDataStreams.Payload1 = complexObjectMultipleDataStreams.Payload1.ConfigureWriterOnReceivedDataStream();
+            complexObjectMultipleDataStreams.Payload2 = complexObjectMultipleDataStreams.Payload2.ConfigureWriterOnReceivedDataStream();
+            return complexObjectMultipleDataStreams;
         }
 
         public ComplexObjectMultipleChildren Process(ComplexObjectMultipleChildren request)
         {
-            request.Child1.ChildPayload1 = request.Child1.ChildPayload1.ConfigureWriterOnReceivedDataStream();
-            request.Child1.ChildPayload2 = request.Child1.ChildPayload2.ConfigureWriterOnReceivedDataStream();
-            request.Child1.ListOfStreams = request.Child1.ListOfStreams.Select(x => x.ConfigureWriterOnReceivedDataStream()).ToList();
-            request.Child2.ComplexPayloadSet = request.Child2.ComplexPayloadSet.Select(x => new ComplexPair<DataStream>(x.EnumValue, x.Payload.ConfigureWriterOnReceivedDataStream())).ToHashSet();
-            request.Child1.ListOfStreams = request.Child1.ListOfStreams.Select(x => x.ConfigureWriterOnReceivedDataStream()).ToList();
-            request.Child2.ComplexPayloadSet = request.Child2.ComplexPayloadSet.Select(x => new ComplexPair<DataStream>(x.EnumValue, x.Payload.ConfigureWriterOnReceivedDataStream())).ToHashSet();
-
-            var response = service.Process(request);
-            
-            response.Child1.ChildPayload1 = response.Child1.ChildPayload1.ConfigureWriterOnReceivedDataStream();
-            response.Child1.ChildPayload2 = response.Child1.ChildPayload2.ConfigureWriterOnReceivedDataStream();
-            response.Child1.ListOfStreams = response.Child1.ListOfStreams.Select(x => x.ConfigureWriterOnReceivedDataStream()).ToList();
-            response.Child2.ComplexPayloadSet = response.Child2.ComplexPayloadSet.Select(x => new ComplexPair<DataStream>(x.EnumValue, x.Payload.ConfigureWriterOnReceivedDataStream())).ToHashSet();
-            response.Child2.ComplexPayloadSet = response.Child2.ComplexPayloadSet.Select(x => new ComplexPair<DataStream>(x.EnumValue, x.Payload.ConfigureWriterOnReceivedDataStream())).ToHashSet();
+            var response = service.Process(ConfigureWriterOnReceivedDataStream(request));
+            return ConfigureWriterOnReceivedDataStream(response);
+        }
 
-            return response;
+        static ComplexObjectMultipleChildren ConfigureWriterOnReceivedDataStream(ComplexObjectMultipleChildren complexObjectMultipleChildren)
+        {
+            complexObjectMultipleChildren.Child1.ChildPayload1 = complexObjectMultipleChildren.Child1.ChildPayload1.ConfigureWriterOnReceivedDataStream();
+            complexObjectMultipleChildren.Child1.ChildPayload2 = complexObjectMultipleChildren.Child1.ChildPayload2.ConfigureWriterOnReceivedDataStream();
+            complexObjectMultipleChildren.Child1.ListOfStreams = complexObjectMultipleChildren.Child1.ListOfStreams.Select(x => x.ConfigureWriterOnReceivedDataStream()).ToList();
+            complexObjectMultipleChildren.Child2.ComplexPayloadSet = complexObjectMultipleChildren.Child2.ComplexPayloadSet.Select(x => new ComplexPair<DataStream>(x.EnumValue, x.Payload.ConfigureWriterOnReceivedDataStream())).ToHashSet();
+            complexObjectMultipleChildren.Child1.ListOfStreams = complexObjectMultipleChildren.Child1.ListOfStreams.Select(x => x.ConfigureWriterOnReceivedDataStream()).ToList();
+            complexObjectMultipleChildren.Child2.ComplexPayloadSet = complexObjectMultipleChildren.Child2.ComplexPayloadSet.Select(x => new ComplexPair<DataStream>(x.EnumValue, x.Payload.ConfigureWriterOnReceivedDataStream())).ToHashSet();
+            return complexObjectMultipleChildren;
         }
     }
 }

```

Reply:
Ah yeah, the request and the response used to be different classes, but I can consolidate now that they use the same class.